modules/reindex/src/main/java/org/elasticsearch/index/reindex/TransportReindexAction.java

```diff
@@ -179,16 +186,25 @@ public AsyncIndexBySearchAction(BulkByScrollTask task, ESLogger logger, ParentTa
         @Override
         protected ScrollableHitSource buildScrollableResultSource(BackoffPolicy backoffPolicy) {
             if (mainRequest.getRemoteInfo() != null) {
-                // NORELEASE track 500-level retries that are builtin to the client
                 RemoteInfo remoteInfo = mainRequest.getRemoteInfo();
+                HttpClientBuilder httpClient = HttpClients.custom();
                 if (remoteInfo.getUsername() != null) {
-                    // NORELEASE support auth
-                    throw new UnsupportedOperationException("Auth is unsupported");
+                    UsernamePasswordCredentials creds = new UsernamePasswordCredentials(remoteInfo.getUsername(), remoteInfo.getPassword());
```

Comment:
I don't think there is much we can do here to really add a lot of value other than try to minimize the time that the password is in plaintext after we are done with it.

We could do this by using a char[] instead of string for the password in the request and clear it out after serialization (assuming we no longer need it on that node). We'd add our own `Credentials` implementation that uses a char[] and clears it once we're done. In the future we could also use a char[] with [httpclient 5.0](https://hc.apache.org/httpcomponents-client-ga/httpclient/apidocs/org/apache/http/auth/UsernamePasswordCredentials.html#UsernamePasswordCredentials(java.lang.String)).

This isn't really foolproof since objects get moved around as part of garbage collection (eden, survivor, old, etc) and there could be copies that aren't cleared until something else gets allocated there. Also, things like request parsing would probably load the value in memory as a regular String, which is out of our control. This relates to #10950. I'd view this as more of a enhancement and not a blocker for this PR